gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java

```diff
@@ -320,33 +319,33 @@ public Single<Certificate> update(String domain, String id, UpdateCertificate up
     @Override
     public Completable delete(String certificateId, User principal) {
         LOGGER.debug("Delete certificate {}", certificateId);
-        return certificateRepository.findById(certificateId)
-                .switchIfEmpty(Maybe.error(new CertificateNotFoundException(certificateId)))
-                .flatMapSingle(certificate -> applicationService.findByCertificate(certificateId).count()
+        return Completable.fromSingle(certificateRepository.findById(certificateId)
+                .switchIfEmpty(Single.error(() -> new CertificateNotFoundException(certificateId)))
+                .flatMap(certificate -> applicationService.findByCertificate(certificateId).count()
                         .flatMap(applications -> {
                             if (applications > 0) {
-                                return Single.error(new CertificateWithApplicationsException());
+                                return Single.error(CertificateWithApplicationsException::new);
                             }
                             return Single.just(certificate);
                         })
                 )
-                .flatMapCompletable(certificate -> {
-                    // create event for sync process
+                .flatMap(certificate -> certificateRepository.delete(certificateId).andThen(Single.just(certificate)))
+                .doOnSuccess(certificate -> {
+                    auditService.report(AuditBuilder.builder(CertificateAuditBuilder.class).principal(principal).type(EventType.CERTIFICATE_DELETED).certificate(certificate));
                     Event event = new Event(Type.CERTIFICATE, new Payload(certificate.getId(), ReferenceType.DOMAIN, certificate.getDomain(), Action.DELETE));
-                    return Completable.fromSingle(certificateRepository.delete(certificateId)
-                            .andThen(eventService.create(event)))
-                            .doOnComplete(() -> auditService.report(AuditBuilder.builder(CertificateAuditBuilder.class).principal(principal).type(EventType.CERTIFICATE_DELETED).certificate(certificate)))
-                            .doOnError(throwable -> auditService.report(AuditBuilder.builder(CertificateAuditBuilder.class).principal(principal).type(EventType.CERTIFICATE_DELETED).throwable(throwable)));
+                    eventService.create(event).subscribe();
```

Comment:
**question**: do we really want to fire&forget the event creation ? 🤔 If an error occures the event may not be persist in DB and so the GW will never see the change. Before this update, an error will be display on the UI, after this change, the UI will display a successful response but the GWs will not see the fact that this certificate has been deleted...